addons/pos_viva_wallet/models/pos_payment_method.py

```diff
@@ -0,0 +1,229 @@
+        return 'https://accounts.vivapayments.com'
+
+    def _viva_wallet_api_get_endpoint(self):
+        if self.viva_wallet_test_mode:
+            return 'https://demo-api.vivapayments.com'
+        return 'https://api.vivapayments.com'
+
+    def _viva_wallet_webhook_get_endpoint(self):
+        if self.viva_wallet_test_mode:
+            return 'https://demo.vivapayments.com'
+        return 'https://www.vivapayments.com'
+
+    def _is_write_forbidden(self, fields):
+        # Allow the modification of these fields even if a pos_session is open
+        whitelisted_fields = {'viva_wallet_bearer_token', 'viva_wallet_webhook_verification_key', 'viva_wallet_latest_response'}
+        return bool(fields - whitelisted_fields and self.open_session_ids)
+
+    def _get_payment_terminal_selection(self):
+        return super()._get_payment_terminal_selection() + [('viva_wallet', 'Viva Wallet')]
+
+    def _get_session(self):
+        session = requests.Session()
+        session.mount('https://', requests.adapters.HTTPAdapter(max_retries=RETRY))
+        session.headers.update({
+            "Connection": "Keep-Alive",
+            "Keep-Alive": "timeout=40, max=120"
+            })
+        return session
+
+    def _bearer_token(self):
+        self.ensure_one()
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Do not have access to fetch token from Viva Wallet"))
+
+        session = requests.Session()
+        session.mount('https://', requests.adapters.HTTPAdapter(max_retries=RETRY))
+        data = {'grant_type': 'client_credentials'}
+        auth = requests.auth.HTTPBasicAuth(self.viva_wallet_client_id, self.viva_wallet_client_secret)
+        try:
+            resp = session.post(f"{self._viva_wallet_account_get_endpoint()}/connect/token", auth=auth, data=data, timeout=TIMEOUT)
+        except requests.exceptions.RequestException:
+            _logger.exception("Failed to call viva_wallet_bearer_token endpoint")
+
+        self.viva_wallet_bearer_token = json.loads(resp.text).get('access_token')
+
+    def _get_verification_key(self, endpoint, viva_wallet_merchant_id, viva_wallet_api_key):
+        # Get a key to configure the webhook.
+        # this key need to be the response when we receive a notifiaction
+        # do not execute this query in test mode
+        if tools.config['test_enable']:
+            return 'viva_wallet_test'
+
+        session = requests.Session()
+        session.mount('https://', requests.adapters.HTTPAdapter(max_retries=RETRY))
+        session.auth = (viva_wallet_merchant_id, viva_wallet_api_key)
+        try:
+            resp = session.get(f"{endpoint}/api/messages/config/token", timeout=TIMEOUT)
+        except requests.exceptions.RequestException:
+            _logger.exception('Failed to call https://%s/api/messages/config/token endpoint', endpoint)
+        return json.loads(resp.text).get('Key')
```

Comment:
```suggestion
        return resp.json().get('Key')
```
(same for other calls below)